### ISSUE TITLE
Simplecov reports lack of coverage in spec files?

### ISSUE THREAD

**Issue by troessner** [troessner/reek#1419]
In the scope of [this pull request](https://github.com/troessner/reek/pull/1382) I added this stub of a future spec:

```Ruby
require_relative '../../spec_helper'
require_lib 'reek/cli/application'

RSpec.describe 'show configuration path' do
  let(:clean_file) do
    <<-EOS
      # Well documented Clean class.
      class Clean
        def hello(argument)
          say argument
        end
      end
    EOS
  end

  around do |example|
    Dir.mktmpdir('/tmp') do |tmp|
      Dir.chdir(tmp) do
        File.write('clean.rb', clean_file)
        example.run
      end
    end
  end
end
```

Surprisingly for me, simplecov fails now with:

```
Coverage report generated for RSpec to /Users/timo/code/reek/coverage. 6724 / 6771 LOC (99.31%) covered.
File (/Users/timo/code/reek/spec/reek/cli/show_configuration_path_spec.rb) is only (50.00%) covered. This is below the expected minimum coverage per file of (81.40%)
```

already which I find very odd. Is this the idea behind this that we write tests for set up code in specs as well? If yes then I'd like to question this configuration and rather have simplecov not cover our specs since this seems like too much to me.

I added a [root](https://www.rubydoc.info/gems/simplecov/SimpleCov%2FConfiguration:root) to the configuration like this:

```Ruby
SimpleCov.start do
  root 'lib/'
  track_files 'lib/**/*.rb'
  add_filter 'lib/reek/version.rb' # version.rb is loaded too early to test
  add_filter 'lib/reek/cli/options.rb' # tested mostly via integration tests
end
```

to prevent simplecov from checking anything outside of lib/ but if I do it like that it seems the filters below are ignored since simplecov fails again:

```
Coverage report generated for RSpec to /Users/timo/code/reek/lib/coverage. 2922 / 2985 LOC (97.89%) covered.
Coverage (97.89%) is below the expected minimum coverage (98.90%).
```

Looking at the coverage report I can see the version.rb up there with 0% coverage, so it seems like adding the root fixed the spec problem but introduced the new problem that our defined filters for the version file are not working anymore.

Is this a problem you have come across already @mvz ? It seems like a simplecov bug to me at the moment but maybe I'm missing something.

**Comment by troessner** [comment]
And a step further:

Making all paths for our filters relative:

```Ruby
SimpleCov.start do
  root 'lib/'
  track_files './**/*.rb'
  add_filter 'reek/version.rb' # version.rb is loaded too early to test
  add_filter 'reek/cli/options.rb' # tested mostly via integration tests
end
```

works in the sense that version.rb and options.rb are not reported anymore. However now I have the problem that simplecov still fails because now with the spec files not checked anymore we run into this:

```
Coverage report generated for RSpec to /Users/timo/code/reek/lib/coverage. 2831 / 2867 LOC (98.74%) covered.
Coverage (98.74%) is below the expected minimum coverage (98.90%).
```

with the top offenders being 

* reek/ast/sexp_extensions/attribute_assignments.rb
* reek/ast/sexp_extensions/literal.rb

So removing the spec files from checking decreased the average coverage since the coverage of our spec files was usually 100%.

Long story short: The original issue has been resolved but I'd like to use this issue to discuss this:

```
Is this the idea behind this that we write tests for set up code in specs as well? If yes then I'd like to question this configuration and rather have simplecov not cover our specs since this seems like too much to me.
```

**Comment by troessner** [comment]
By the way, this

```

```

doesn't seem to make any difference for our code base. On master this gives us

```
214 files in total. 6751 relevant lines. 6714 lines covered and 37 lines missed
```
with that line commented out or not and on the "show-configuration-path" feature branch (using `root`):

```
130 files in total. 2867 relevant lines. 2831 lines covered and 36 lines missed
```

so it seems safe for us to drop it.

**Comment by mvz** [comment]
I don't know why SimpleCov considers code in specs when calculating coverage. That was never the intent anyway, but indeed I do see files in `spec/` appearing in the coverage report.

**Comment by mvz** [comment]
The reason for `track_files` is that we want to detect the case where an entire file is never included. In such a case, SimpleCov won't know it exists unless told through `track_files`.
